Test for partitioner failure

twsearle opened the pull request:
## Description

My application as well as `atlas-meshgen` are reporting errors in `BuildParallelFields.cc` with orca grids when using multiple processors.

I think this issue has something to do with `atlas-orca`, but I am not sure. I haven't made much progress identifying the source of the issue, but these are some tests to begin to understand what is going wrong.

When running `mpirun -np 16 ./bin/atlas-meshgen ORCA2_T` I am seeing errors like:
```
[0] Node requested by rank [7] with uid [940563842883868512] that should be owned is not found
```
or many messages of the form:
```
[0] Node with global index 12923 not found on part [7]
```
My first instinct was that somehow the partition is not getting set properly by the mesh generator, although I admit I am still a bit confused about how everything fits together! I have written a few tests. 

So far I can't reproduce the errors above in my tests, however, I do find that calling the partitioner within `meshgenerator.generate` will let me run the `checkerboard` partitioner, but using `partitioner.partition(grid)` will throw the following exception:
```
"test orca available partitioners (section: ORCA2_T checkerboard)" failed with unhandled eckit::Exception: Checkerboard Partitioner only works for Regular grids.
```

I am still experimenting with this, hopefully I will be able to add more information via further tests here.

twsearle committed:
Add tests of the partitioner on 2 processors

wdeconinck commented:
I think the original issue is solved with commits to develop on both atlas and atlas-orca.
atlas: https://github.com/ecmwf/atlas/commit/4150608c4dc71932ce1ebe168d9ba8189c75f609
atlas-orca: https://github.com/wdeconinck/atlas-orca/commit/c663e604c75cf7636787b94efa9911708b4332ce

twsearle committed:
Merge branch 'develop' of github.com:twsearle/atlas-orca into partitioner-tests

twsearle committed:
Remove partitioner.partition tests

This doesn't work because checkerboard requires a regular grid. In the
other constructors, a regular grid is built in the
meshgenerator.generate function before being handed to the partitioner

twsearle committed:
tests: interpolation tests

twsearle committed:
test adding halo's during mesh generation
